src/Nethermind/Nethermind.Merge.Plugin/MergeGossipPolicy.cs

```diff
@@ -36,7 +36,8 @@ public class MergeGossipPolicy : IGossipPolicy
         // According to spec (https://github.com/ethereum/EIPs/blob/d896145678bd65d3eafd8749690c1b5228875c39/EIPS/eip-3675.md#network)
         // We MUST discard NewBlock/NewBlockHash messages after receiving FIRST_FINALIZED_BLOCK.
         public bool ShouldDiscardBlocks => _poSSwitcher.TransitionFinished ||
-                                           _blockCacheService.FinalizedHash != Keccak.Zero;  /* _blockCacheService.FinalizedHash != Keccak.Zero - this condition was added for edge case situation.
+                                           (_blockCacheService.FinalizedHash != null && _blockCacheService.FinalizedHash != Keccak.Zero);  /*  _blockCacheService.FinalizedHash != null && _blockCacheService.FinalizedHash != Keccak.Zero
```

Comment:
1. `is not null`
2. Maybe we can have a method `IsNotNullOrZero` like `string.IsNotNullOrEmpty`